### DISCUSSION TITLE
How can I get the final value of all the metrics that have been logged

### DISCUSSION THREAD

**Discussion by pamparana34** [Lightning-AI/pytorch-lightning#12735] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
At the moment, I am calling `trainer.fit(model=model, datamodule=datamodule)` and the training works fine.
However, what I want to do is return the final value of all the metrics that have been logged during the training.

In my case, the training and validation steps looks as follows:

```
def validation_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
        loss = torch.stack(outputs).mean()
        self.log("val/loss", loss)

def training_epoch_end(self, outputs: EPOCH_OUTPUT) -> None:
        loss = torch.stack(outputs).mean()
        self.log("train/loss", loss)
```

Basically, at the end of the training, I would like to query the final values of `train/loss` and `val/loss` and basically every other metrics that have been logged.

**Comment by rohitgr7** [comment, marked as answer]
`trainer.callback_metrics` <- to access all the metrics
`trainer.logged_metrics` <- to access only the logged metrics
